src/components/AvailableProduct/productPage.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState, AppDispatch } from '../../redux/store';
+import { fetchAvailableProducts } from '../../redux/slices/availableProductSlice';
+import Product from '../AvailableProduct/product';
+import Pagination from '../AvailableProduct/pagination';
+
+const ProductsPage: React.FC = () => {
+  const dispatch: AppDispatch = useDispatch();
+  const { products, loading, error, totalPages, currentPage: reduxCurrentPage } = useSelector((state: RootState) => state.products);
+
+  const [currentPage, setCurrentPage] = useState(reduxCurrentPage); 
+  const productsPerPage = 10;
+
+  useEffect(() => {
+    dispatch(fetchAvailableProducts(currentPage));
+  }, [dispatch, currentPage]);
+
+  const handlePageChange = (pageNumber: number) => {
+    setCurrentPage(pageNumber); 
+  };
+
+  if (loading) return <div>Loading...</div>;
```

Comment:
Can you render this `Loading ....` under the Product title? Otherwise, someone might need clarification about which data are being loaded.